### ISSUE TITLE
Accessibility issues with JUnit

### ISSUE THREAD

**Issue by SamKacer** [albzan/eclipse-nvda#2]
So, the add on is really great and solves a bunch of things that really bugged me in Eclipse before I discovered this addon, but one big issue I still have with using Eclipse is that as far as I can tell JUnit is pretty much useless for a screen reader user, as whether a test passed or failed is complletely shown visually (highlighted green is pass, highlighted red is fail, i think).

I am a big fan of test driven development, so I really wish this issue could be fixed through a NVDA addon.

Would it be possible for this addon to extend its functionality to report passes and fails when reviewing the JUnit report in Eclipse.

Otherwise is there any advice for using JUnit testing in Eclipse as a screen reader user?

Thanks.

**Comment by kaiyasa** [comment]
I realize this has been open for a long time..  However, I may be able to help.

I'm using NVDA with eclipse (Oxygen) and routinely use jUnits as well.  The base accessibility will work with a bit of tweaking.  Using ControlF7 until you select the jUnit view.  The Shift-Tab about 4-5 times until you find the View menu.  Activate it (I use space bar) and Down arrow until you find "Show only failed tests" and activate again (space bar).  This will limit to only the failed tests now.  Once in the tree view again, use up and down arrows to hear test names if any.  Once you found one of interest, hit Tab twice to get the stack trace info.. You can use up and down arrows again to find a frame and hit Enter to jump to line and switch to the editor conveniently.  On a side note, I use hamcrest matchers which make life much easier without a debugger.

I came looking to make debugging accessible and thus far have failed to get this addon working on Oxygen.

**Comment by albzan** [comment]
Hi,
I use the object navigation on the JUnit Window to look how many tests pass and how many fails. When in the test list, move to the previous object twice to see Failed and Errors labels.
Cheers,
Alberto
